File: AlcoholDrive/src/app/domain/repositories/UserRepository/User.repository.ts
import { Injectable } from "@angular/core";
import { UserModel } from "../../model/User.model";

@Injectable({
    providedIn: 'root'
})
export abstract class UserRepository {
    /**ユーザを取得 */
    abstract getUsers(): Promise<void>
    /**ユーザを登録 */
    abstract registryUser(user: UserModel): Promise<void>;
    /**ユーザを削除 */
    abstract deleteUser(userId: number): Promise<void>;

    /** 上長を設定する */
    abstract setUserBoss(userId: number, bossId: number): Promise<void>;
}

File: AlcoholDrive-Client/AlcoholDrive-Client/Repository/UserRepository.cs
using AlcoholDrive_Client.Model;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace AlcoholDrive_Client.Repository {

    public abstract class UserRepository {

        /// <summary>
        /// ユーザ一覧を返す
        /// </summary>
        /// <returns></returns>
        public abstract List<UserModel> GetUsers();
        /// <summary>
        /// ユーザを登録する
        /// </summary>
        /// <param name="user"></param>
        public abstract void RegistryUser(UserModel user);
        /// <summary>
        /// ユーザを削除する
        /// </summary>
        /// <param name="userId"></param>
        public abstract void DeleteUser(int userId);
        /// <summary>
        /// 上長を設定する
        /// </summary>
        /// <param name="userId"></param>
        /// <param name="bossId"></param>
        public abstract void SetUserBoss(int userId, int bossId);
    }
}
